backend/src/services/trading_service.py

```
@@ -105,6 +105,29 @@ async def _get_swap_instructions(self, input_mint: str, output_mint: str, amount
             logger.error(f"Unexpected error in _get_swap_instructions: {e}")
             return None
 
+    async def _confirm_transaction_robustly(self, tx_signature: Signature, max_retries: int = 30, delay: float = 2.0) -> bool:
```

Ruff (0.15.2)

[error] Undefined name `Signature`

(F821)